src/main/java/com/quantumsoft/qupathcloud/dao/spec/FramePathBuilder.java

```diff
@@ -15,43 +15,62 @@
 
 package com.quantumsoft.qupathcloud.dao.spec;
 
+import static com.quantumsoft.qupathcloud.dao.Constants.DATASETS;
+import static com.quantumsoft.qupathcloud.dao.Constants.DICOM_STORES;
+import static com.quantumsoft.qupathcloud.dao.Constants.DICOM_WEB;
+import static com.quantumsoft.qupathcloud.dao.Constants.FRAMES;
+import static com.quantumsoft.qupathcloud.dao.Constants.INSTANCES;
+import static com.quantumsoft.qupathcloud.dao.Constants.LOCATIONS;
+import static com.quantumsoft.qupathcloud.dao.Constants.PROJECTS;
+import static com.quantumsoft.qupathcloud.dao.Constants.SERIES;
+import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
+import static com.quantumsoft.qupathcloud.dao.Constants.STUDIES;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * PFramePathBuilder to build a path for access to a frame in chosen Project, Location, Dataset,
+ * DICOM Store, Study, Series, and Instance.
```

Comment:
PFramePathBuilder builds a path to access a frame in the specified Project, Location, Dataset, DICOM Store, Study, Series, and Instance.